packages/web-fragments/src/gateway/fragment-gateway.ts

```diff
@@ -27,15 +27,29 @@ export interface FragmentConfig {
    * This will be fetched on any request paths matching the specified `routePatterns`
    */
   upstream: string;
+  /**
+   * Handler/Fallback to apply when the fetch for a fragment ssr code fails.
+   * It allows the gateway to serve the provided fallback response instead of an error response straight
+   * from the server.
+   *
+   * @param req the request sent to the fragment
+   * @param failedRes the failed response (with a 4xx or 5xx status) or the thrown error
+   * @returns the response to use for the document's ssr
+   */
+  onFragmentFailedFetch?: (req: RequestInfo, failedResOrError: Response|unknown) => Response|Promise<Response>;
```

Comment:
maybe, I wanted to make it clear that this is only for the fragment ssr fetch though...

what about `onSsrFetchError`?

(PS: since I refactored it and it is in the fragment config you're right that the fragment word is totally unnecessary 🤔 )